### PR TITLE
MFB-452: update immigration filters

### PR DESCRIPTION
**Pull request by catonph**
## Context & Motivation

The citizenship filter previously used a multi-select boolean-based structure (`Record<CitizenLabels, boolean>`) with 15 separate boolean flags. This approach had several limitations:
- Complex state management with multiple flags being toggled simultaneously
- Difficult to understand which single citizenship status was actually "selected"
- Required conversion functions to derive actual filter state
- Made it unclear that only one citizenship status should be active at a time

This PR refactors the citizenship filter to use a **single-select model** with a new `FilterState` type that explicitly represents one selected citizenship and its derived/calculated filters.

BE PR: https://github.com/MyFriendBen/benefits-api/pull/1268
Ticket: https://linear.app/myfriendben/issue/MFB-452/general-improve-immigration-filter-dropdown-ui

### Filter State Restructuring
- **Created new `FilterState` type** with two properties:
  - `selectedCitizenship`: The single citizenship option currently selected
  - `calculatedFilters`: A `Set<CitizenLabelOptions>` containing derived filters based on household data
- **Removed deprecated types**: `FiltersChecked` type and conversion functions (`citizenshipFiltersToFiltersChecked`, `filtersCheckedToCitizenshipFilters`)
- **Updated `calculateDerivedFilters` function** to return a `Set` instead of an array for more efficient membership testing

### Component Updates
- **Filter.tsx**: Updated to work with new single-select `FilterState`
- **Results.tsx**: 
  - Initialize `filterState` with single-select model
  - Removed deprecated conversion logic
- **ProgramPage.tsx**: Updated warning message filtering to check both `selectedCitizenship` and `calculatedFilters` Set
- **filterPrograms.ts**: Updated eligibility logic to use `calculatedFilters` Set

### Code Quality
- **Comprehensive test coverage**: Added 32 unit tests for Filter.tsx covering:
  - Component rendering (desktop buttons vs mobile dropdown)
  - Button and dropdown interactions
  - Calculated filters integration with household data
  - Description collapse/expand functionality
  - Accessibility (ARIA labels, roles, live regions)
  - Responsive behavior
  - Edge cases and integration with FilterState
- **Simplified state management**: Single source of truth for selected citizenship
- **Improved type safety**: Explicit types make it clear only one citizenship can be selected at a time

image
image
image
image

### Manual Testing Steps
1. Navigate to the results page with household data that includes multiple citizenship statuses
2. Verify the citizenship filter displays with "U.S. Citizen" selected by default
3. Click different citizenship buttons (desktop) or select from dropdown (mobile) and verify:
   - Selected option updates correctly
   - Programs filter appropriately based on selection
   - Warning messages on program pages respect both selected and calculated filters
4. Test with different household compositions to verify calculated filters work correctly (e.g., children with different statuses than adults)
5. Verify responsive behavior at 768px threshold (desktop/mobile breakpoint)
6. Test description collapse/expand on narrow screens (≤730px)

### Automated Testing
Run the test suite:
npm test Filter.test.tsx
All 32 tests should pass, covering rendering, interactions, accessibility, and edge cases.

## Deployment

*Already* added to staging/prod:

**1. Citizenship Button Labels (6 new IDs)**
- `citizenshipButton-citizen` - "U.S. Citizen"
- `citizenshipButton-gc_5plus` - "Green Card 5+"
- `citizenshipButton-gc_5less` - "Green Card <5"
- `citizenshipButton-refugee` - "Refugee/Asylee"
- `citizenshipButton-otherLawful` - "Other Lawful"
- `citizenshipButton-undocumented` - "Undocumented"

**2. Citizenship Tooltip Text (6 new IDs)**
- `citizenshipTooltip-citizen` - "U.S. citizens by birth or naturalization."
- `citizenshipTooltip-gc_5plus` - "Lawful permanent residents who have had their green card for 5 or more years."
- `citizenshipTooltip-gc_5less` - "Lawful permanent residents who have had their green card for less than 5 years."
- `citizenshipTooltip-refugee` - "Individuals granted refugee or asylee status by the U.S. government."
- `citizenshipTooltip-otherLawful` - "Other lawfully present noncitizens with authorization to live or work in the U.S."
- `citizenshipTooltip-undocumented` - "Individuals without lawful immigration status (includes DACA recipients)."

**3. Accessibility ARIA Labels (7 new IDs)**
- `citizenshipButton-citizen-aria` - "Select citizen citizenship status"
- `citizenshipButton-gc_5plus-aria` - "Select gc_5plus citizenship status"
- `citizenshipButton-gc_5less-aria` - "Select gc_5less citizenship status"
- `citizenshipButton-refugee-aria` - "Select refugee citizenship status"
- `citizenshipButton-otherWithWorkPermission-aria` - "Select otherWithWorkPermission citizenship status"
- `citizenshipButton-non_citizen-aria` - "Select non_citizen citizenship status"
- `citizenshipDropdown-aria` - "Select citizenship status"

**4. Updated Existing IDs**
- `filterSection.header` - "Filter Results by Citizenship"
- `filterSection.citizenHelpText` - "Use this filter to see which benefits a person might qualify for based on their immigration status. If people in your home have different statuses, you can check each person one at a time. Your information is private—we don't save or share anything you choose here."
- `filterSection.showMore` - "Show more"
- `filterSection.showLess` - "Show less"
- `filterSection.currentSelection` - "Current selection:"

### Type Changes (Breaking)
The following types have been updated or removed:
- **Removed**: `FiltersChecked` type
- **Removed**: `citizenshipFiltersToFiltersChecked()` helper
- **Removed**: `filtersCheckedToCitizenshipFilters()` helper
- **Added**: `FilterState` type with `selectedCitizenship` and `calculatedFilters`

### Breaking Changes
- The `FiltersChecked` type and its conversion helpers have been completely removed
- The filter state is now initialized with a single selected citizenship instead of multiple boolean flags

### Key Architectural Changes
- **Single-select model**: Only one citizenship option is "selected" at a time, with calculated filters derived automatically
- **Set-based derived filters**: Using `Set<CitizenLabelOptions>` for `calculatedFilters` provides O(1) membership testing
- **Clearer intent**: The new structure makes it explicit that users are selecting a single citizenship status to view results for that person

### Testing Approach
- The test suite focuses on testing the component contract (rendering, interactions, accessibility) rather than implementation details
- State change testing for simple UI features (expand/collapse) was intentionally kept minimal to avoid test complexity overhead
- Tests use proper mocking for all dependencies (Results context, Wrapper context, Material-UI, react-intl)

### CODE REVIEW

**Comment by catonph** [comment]
@coderabbitai review

File: .github/workflows/tests-and-coverage.yml [hunk 1]
```diff
@@ -3,6 +3,9 @@ name: Tests & Coverage
 on:
   pull_request:
     branches: [ main ]
+  push:
+    branches:
+      - '**'  # Run on pushes to all branches
```
Comment (catonph) [inline .github/workflows/tests-and-coverage.yml]: want codecov to rerun when we push (right now just does the report once)

File: src/Components/Results/Filter/citizenshipFilterConfig.tsx [hunk 2]
```diff
@@ -0,0 +1,198 @@
+import { FormattedMessage } from 'react-intl';
```
[the remark this answers is not in the document]
Comment (catonph) [reply src/Components/Results/Filter/citizenshipFilterConfig.tsx]: Good to know! TY
